body.tex

```diff
@@ -1,166 +1,466 @@
+
+Compute resources are procured from the master node via \texttt{condor\_annex} by targeting the worker AMI. Annex is a resource acquisition and external compute resource separation management unit. Each resource pool can have multiple annexes. Each annex manages its own lifecycle. It is possible to request on-demand or spot instance fleets.
+
+LSST Pipeline jobs are executed on the worker instances; they use Data Butler for data access.
+New Data Butler backends were implemented during the PoC, see Sect \ref{sec:butler}.
 
+The Butler datastore is located in an S3 Bucket and follows the same hierarchical structure that POSIX datastore does. Consumed and produced datasets are read and written directly from S3 as bytes, whenever possible, and only downloaded to temporary files for objects that are not serializable.
+Since the directory structure is preserved by the S3 datastore the entire data repository is trivially transferable between the cloud and a local filesystem.
 
+The Butler registry is a RDS PostgreSQL database that keeps track of all LSST science files.
+Files that are not managed by Data Butler are managed by the Condor File IO via Pegasus. These include the Data Butler configuration file, pipeline definition (Quantum) files, and the log files. The master instance serves as the staging site for these files. Various network protocols transfer the files between the master and the workers; instances do not share a filesystem.
 
+\subsection{Alternative architecture designs}
 
-\section{Some technicalities: author list and standard LSST references} 
+We discussed different architecture designs but did not pursue all of them due to time constraints.
 
-Thank you Tim Jenness and Wil O'Mullane for helping with templates! 
+One prominent idea was to use HTCondor utilities to transfer files with S3, rather than relying on Data Butler to communicate directly with S3. In this design HTCondor controls all file transfers, and all datasets are declared in the execution workflow. The proposed implementation was partially external to the LSST code base, and such a data transfer plug-in was added to the HTCondor code base recently. In each job there is a POSIX data repository for local data access, and a mechanism is needed to trigger and perform data ingestion at a central S3 data repository. This requires additional Generation 3 Middleware utilities for posix-S3 datastore copying and URL generation. These issues disfavored this architecture. A favorable benefit of this approach is the potential of bulk data transfer management optimizations.
 
-\subsection{The LSST LaTeX Classes}
+With respect to the database, we discussed the possibility of having local registries within individual jobs.
+Each job has a fast local SQLite registry.
+After each job finishes, the individual registry is then collated into the central database.
```

Comment:
Add "This is a subset of the previous design alternative, with only the registry being localized and not the datasets."

Reply:
Fixed but left open. Resolved.